### PR TITLE
Embed ui

### LINKED ISSUE #2182: Embed map
- [x] create embed.html

Create interface
- [x] add Text "Embed map" with one line with default code and disclaimer below
- [x] add show More options above disclaimer
- [x] "Show more" modal window contains
 Pulldown with
    - [x] preset sizes
    - [x] Custom sizes (minimum 200x200px), no max size limit
    - [x] Preview map (adapting to size)
- [x] disclaimer "By embedding this map, you agree to the terms of service" with link to http://www.geo.admin.ch/internet/geoportal/en/home/geoadmin/contact.html#iFrame

Embed map
- [x] white Box (analogue gmap embed) with top left, transparent outside: it is a link it goes to the map.geo.admin.ch the full size application (deskop mode)
 - [x]   icon (Favicon ) 
 - [x]   Text "view on map.xxx.admin.ch" xxx replaced by source domain
 Icons 
- [x] no geolocation
- [x] no overview
- [x] Footer left keep scale ( max size to middle of width) 
- [x] Footer right "C Data:" as in mobile ( max size to middle of width) if bigger then add 3 points (...) on mouseover show all dataowners links etc
- [ ] detect if embed.html is used in an iframe or embed context, if not display only a text "The map.geo.admin.ch Maps Embed must be used in an iframe." (Analogue to Google maps embed)

### PR DESCRIPTION
**Pull request by oterral**
Add ui components for embed page

Fix #2182 

[Test](http://mf-geoadmin3.dev.bgdi.ch/teo_embed_ui/?lang=fr&topic=ech&bgLayer=ch.swisstopo.pixelkarte-farbe&X=190500.00&Y=660000.00&zoom=1)

### CODE REVIEW

**Comment by davidoesch** [comment]
Proposal
the "more ..." button

remove it

and add a new line with an URL : "Show more"  (analogue youtube)

**Comment by cedricmoullet** [comment]
just playing with it. just great !

On Tue, Mar 10, 2015 at 9:02 AM, davidoesch <notifications@github.com>
wrote:

> Proposal
> the "more ..." button
>
> remove it
>
> and add a new line with an URL : "Show more" (analogue youtube)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geoadmin/mf-geoadmin3/pull/2204#issuecomment-78010061>
> .
>



-- 
Twitter: http://twitter.com/cedricmoullet
Linked In: http://www.linkedin.com/in/cedricmoullet

<http://t.signaletre.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJN7t5XYgf6M1WN1qfVVMRYFRqW8qlQyH56dByMf1q8vLz02?t=http%3A%2F%2Ftwitter.com%2Fcedricmoullet&si=6057645868318720&pi=8f475c3d-cdef-42a5-c8eb-d689c49114c5>

**Comment by davidoesch** [comment]
the link for the terms and conditions is

www.geo.admin.ch/internet/geoportal/de/home/geoadmin/contact.html#iFrame
www.geo.admin.ch/internet/geoportal/fr/home/geoadmin/contact.html#iFrame
www.geo.admin.ch/internet/geoportal/it/home/geoadmin/contact.html#iFrame
www.geo.admin.ch/internet/geoportal/en/home/geoadmin/contact.html#iFrame
www.geo.admin.ch/internet/geoportal/rm/home/geoadmin/contact.html#iFrame

**Comment by davidoesch** [comment]
save space and do it like google (KISS)

replace "Largeur": FIELD  "Hauteur:" FIELD 

with

FIELD x FIELD

**Comment by gjn** [comment]
jenkins test

File: src/components/permalink/PermalinkService.js [hunk 1]
```diff
@@ -49,6 +49,24 @@
...
 
+          this.getEmbedHref = function(p) {
+            var newParams = angular.extend({}, params);
+            if (angular.isDefined(p)) {
+              angular.extend(newParams, p);
+            }
+            if (angular.isDefined(newParams.mobile)) {
+              delete newParams.mobile;
+            }
+            var baseEmbed = base.replace(/(index|mobile)/, 'embed');
```
Comment (gjn) [inline src/components/permalink/PermalinkService.js]: add .html in regex?

File: src/components/permalink/PermalinkService.js [hunk 2]
```diff
@@ -49,6 +49,24 @@
...
+            if (!/embed.html$/.test(baseEmbed)) {
```
Comment (gjn) [inline src/components/permalink/PermalinkService.js]: escape dot

**Comment by gjn** [comment]
Very nice stuff. Probably my most preferred feature in a long time :=)

Maybe it might be worth to put the iframe stuff into separate component (own directive), but given time constraint now, this would be for later (open an issue?)

Question: why is inline preview not possible with custom size settings?

**Comment by davidoesch** [comment]
"Question: why is inline preview not possible with custom size settings?"

it is: you have to add first the dimension... you mean on the fly setting
window size?

**Comment by gjn** [comment]
When in the 'more' window, I can choose 3 pre-defined sizes which are directly displayed in the modal window. When I choose 'personalized', I have to click a link to open a seperate window for the preview. My question was if there is a technical limitation not to have it in the modal window as for the pre-defined sizes.

**Comment by oterral** [comment]
Because there was a lot of css from a lot of different bootstrap element to take in account. It was a mess to manage so I prefered to put a google-like interface which works. We'll do that in a different PR if you really want that.

**Comment by oterral** [comment]
@gjn comments fixed

**Comment by davidoesch** [comment]
as it is now it works like in google.... and it makes sense: you have to
first add the dimensions before showing the map. this si why it makes sense
to start with an empty field

**Comment by gjn** [comment]
Hmm. Nothing hinders us to improve google stuff :=) . I think it's awkward to have different preview for pre-defined stuff and user defined stuff.

Anyhow, @oterral I think @davidoesch wants more (see mail).

**Comment by gjn** [comment]
jenkins failure because of browserstack hicups...

**Pull request merged**